torch/_inductor/scheduler.py

```diff
@@ -1916,6 +1943,145 @@ def create_scheduler_node(self, node: ir.Operation) -> BaseSchedulerNode:
+                        NopKernelSchedulerNode,
+                    ),
+                ):
+                    continue
+                if snode.get_device() != other_node.get_device():
+                    continue
+
+                shared_data_score = self.score_fusion_memory(snode, other_node)
+                if shared_data_score < config.score_fusion_memory_threshold:
+                    shared_data_score = self.shared_data_after_reordering_loop(
+                        snode, other_node
+                    )
+                if shared_data_score >= config.score_fusion_memory_threshold:
+                    _, (numel1, rnumel1) = other_node.group
+                    from .codegen.simd import SIMDScheduling
+
+                    tiling = SIMDScheduling.select_tiling(
+                        other_node.get_nodes(), numel1, rnumel1
+                    )
+                    split_hints.append(tiling[0])
+
+            orig_num_nodes = len(V.graph.operations)
+            orig_buffer = snode.node
+            orig_node = orig_buffer.data
+
+            new_node = ir.Reduction.create(
+                device=orig_node.device,
+                dst_dtype=orig_node.dtype,
+                src_dtype=orig_node.dtype,
+                inner_fn=orig_node.inner_fn,
+                ranges=orig_node.ranges,
+                reduction_ranges=orig_node.reduction_ranges,
+                reduction_type=orig_node.reduction_type,
+                # reduction_hint=orig_node.reduction_hint, # Use default value
+                input_node=orig_node.input_node,
+                ignore_defer_reduction_split=True,
+                split_hints=split_hints,
+            )
+
+            new_node_name = new_node.realize()
+            new_buffer = V.graph.name_to_buffer[new_node_name]
+            assert isinstance(orig_buffer, ir.ComputedBuffer)  # linter
+            assert isinstance(new_buffer, ir.ComputedBuffer)  # linter
+            self.replace_operation_buffer(orig_buffer, new_buffer)
+
+            # After creating the Reduction node, its layout is fixed by
+            # https://github.com/pytorch/pytorch/blob/main/torch/_inductor/graph.py#L1537
+            # Also call `.as_fixed()` to all new reduction nodes.
+            new_buffer.layout = new_buffer.layout.as_fixed()
+            for i in range(orig_num_nodes, len(V.graph.operations)):
+                middle_buffer = V.graph.operations[i]
+                assert isinstance(middle_buffer, ir.ComputedBuffer)  # linter
+                middle_buffer.layout = middle_buffer.layout.as_fixed()
+
+            # Add the new reduction nodes to the scheduler
+            new_scheduler_node = self.create_scheduler_node(new_buffer)
+            self.nodes[idx] = new_scheduler_node
+            self.name_to_node[snode.get_name()] = new_scheduler_node
+            self.name_to_fused_node[snode.get_name()] = new_scheduler_node
+
+            # It's possible that there is no additional middle-layer nodes
```

Comment:
meaning, we didn't split the reduction ? Can we check if there would be a split prior to creating a new reduction ?